src/plugins/share/public/services/share_menu_registry.ts

```diff
@@ -36,7 +36,7 @@ export class ShareMenuRegistry {
     return {
       getShareMenuItems: (context: ShareContext) =>
         Array.from(this.shareMenuProviders.values()).flatMap((shareActionProvider) =>
-          shareActionProvider.getShareMenuItems(context)
+          (shareActionProvider as ShareMenuProviderV2).getShareMenuItems(context)
```

Comment:
This causes an error to be thrown when kibana is started with the `--run-examples` arg in dev, because the share menu of the demo plugin is registered to be retrieved using `getShareMenuItemsLegacy`, So maybe the following would be more appropriate

```suggestion
(shareActionProvider.getShareMenuItems ?? shareActionProvider.getShareMenuItemsLegacy).call(shareActionProvider, context)
```

this way we are inclusive of the fact that some menu item might be defined still using the legacy method

Reply:
@eokoneyo I've made a change that addresses the error: 5b0952f50d2

Unfortunately, registering menu items is broken in Kibana. It looks like this was last working in 8.14

![image](https://github.com/user-attachments/assets/d488d60c-71e8-4fc4-84de-c139312b32b7)